test/hotspot/jtreg/compiler/c2/irTests/TestBoolNodeGVN.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2024 Red Hat and/or its affiliates. All rights reserved.
```

Comment:
Can you move this to a different directory, please? the `irTests` directory was a good idea when we were just starting with the IR framework, but now it makes more sense to sort tests by "what" is tested, rather than "how". Feel free to put it in the `c2` directory, or even create a new subdirectory like `c2/gvn`.